src/main/java/org/squbs/sample/http/SampleHttpSvc.java

```diff
@@ -0,0 +1,100 @@
+package org.squbs.sample.http;
+
+import akka.NotUsed;
+import akka.http.javadsl.marshallers.jackson.Jackson;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import akka.util.Timeout;
+import org.squbs.sample.app.AppConstants;
+import org.squbs.sample.app.ChunkRequest;
+import org.squbs.sample.app.PingRequest;
+import org.squbs.sample.app.PingResponse;
+import org.squbs.unicomplex.AbstractRouteDefinition;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import static akka.http.javadsl.server.PathMatchers.segment;
+import static akka.pattern.PatternsCS.ask;
+
+/**
+ * The route definition.
+ */
+public class SampleHttpSvc extends AbstractRouteDefinition {
+
+    Timeout timeout = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+
+    @Override
+    public Route route() {
+
+        return route(
+           get(() -> route(anonymous, withName, chunks, route(complete(StatusCodes.BAD_REQUEST)))),
+           post(() -> path("hello", () ->
+                entity(Jackson.unmarshaller(PingRequest.class), request -> {
+
+                CompletionStage<PingResponse> respCS =
+                    ask(context().actorSelection(AppConstants.pingActorPath),
+                        request, timeout)
+                        .thenApplyAsync(PingResponse.class::cast);
+
+                return completeOK(respCS.thenApply(o -> o.message), Jackson.marshaller());
```

Comment:
We have to be very judicious when applying abstractions. We should only have `completeOK` if the Akka HTTP API is really weak here. I need to see what it abstracts behind to give a better opinion whether we should abstract in this case. But please consider how much more difficult it would be if we don't abstract.